alacritty_terminal/src/tty/unix.rs

```diff
@@ -241,21 +239,20 @@ pub fn from_fd(config: &Options, window_id: u64, master: OwnedFd, slave: OwnedFd
     builder.env_remove("XDG_ACTIVATION_TOKEN");
     builder.env_remove("DESKTOP_STARTUP_ID");
 
-    let working_directory = config.working_directory.clone();
+    // Set working directory, ignoring invalid paths.
+    if let Some(working_directory) = &config.working_directory {
+        builder.current_dir(working_directory);
```

Comment:
I'm not sure I want to switch to `builder.current_dir` here. I'd have to test it some more but I'd imagine there's probably a reason it wasn't done this way before.

Reply:
The only issue that comes to mind is not being able to start at all without said directory, which we try to avoid.

Reply:
That makes sense, I can replace it with `libc::chdir` as I initially planned to.